modules/elastic-client/src/main/scala/ch/epfl/bluebrain/nexus/commons/es/client/ElasticClient.scala

```diff
@@ -54,15 +54,27 @@ class ElasticClient[F[_]](base: Uri, queryClient: ElasticQueryClient[F])(implici
     *
     * @param index   the index
     * @param payload the payload to attach to the index when it does not exist
-    * @return ''true'' wrapped in ''F'' when index has been created and ''false'' wrapped and ''F'' when it already existed
+    * @return ''true'' wrapped in ''F'' when index has been created and ''false'' wrapped in ''F'' when it already existed
     */
-  def createIndexIfNotExist(index: String, payload: Json): F[Boolean] = {
+  def createIndexIfNotExist(index: String, payload: Json): F[Boolean] =
     existsIndex(index).map(_ => false).recoverWith {
       case ElasticClientError(StatusCodes.NotFound, _) =>
         createIndex(index, payload).map(_ => true)
       case other => F.raiseError(other)
     }
-  }
+
+  /**
+    * Deletes an index when it does not exist on the ElasticSearch endpoint.
+    *
+    * @param index the index
+    * @return ''true'' when the index has been deleted and ''false'' when the index does not exist. The response is wrapped in an effect type ''F''
+    */
+  def deleteIndexIfNotExist(index: String): F[Boolean] =
```

Comment:
I'd rename it to simply: `def deleteIndex(index: String): F[Boolean]` and update the comment to:
_Attempts to delete an index recovering gracefully when the index is not found._

I'd also recommend renaming the `createIndexIfNotExist` function to `def createIndex(index: String, value: Json): F[Boolean]` and replace the description to something similar:
_Attempts to create an index recovering gracefully when the index already exists. Note: it does not check whether the argument index value matches the already created index._